app/src/main/java/com/example/farmeraid/data/QuotasRepository.kt

```diff
@@ -1,115 +1,78 @@
-            ProduceQuota(
-                produceName = "Mango",
-                produceGoalAmount = 5,
-            ),
-        )
-    ), Quota(
-        id = "4",
-        produceQuotaList = listOf(
-            ProduceQuota(
-                produceName = "Banana",
-                produceGoalAmount = 5,
-            ),
-            ProduceQuota(
-                produceName = "Mango",
-                produceGoalAmount = 12,
-            ),
-        )
-    ),
+    //Todo: Refactor to not mock sales data
+    private var SALES = hashMapOf(
+        "mangoes" to 4,
+        "apple" to 2,
+        "strawberry" to 7,
+        "bananas" to 2,
+        "strawberry" to 4
     )
 
-    fun getCategorizedQuotas(): Flow<MutableList<Quota>> {
-        return flow {
-            emit(quotasList)
-        }.flowOn(Dispatchers.IO)
-    }
+    suspend fun getQuota(id: String): Quota? {
+        val docRef = db.collection("quotas").document(id)
 
-    fun getQuota(id : String): Quota? {
-        return quotasList.firstOrNull { quota -> quota.id == id }
-    }
+        val quotas = docRef.get().await()?.data?.get("produce")
+        val saleCount = docRef.get().await()?.data?.get("sale")
+        val produceList: MutableList<ProduceQuota> = mutableListOf()
 
-    fun addQuota(market : MarketModel.Market, produce : List<ProduceQuota>) : String? {
-        val quotaIndex : Int = quotasList.indexOfFirst { quota -> quota.id == market.quotaId }
+        if (quotas != null && saleCount != null) {
+            val quotasMap = quotas as MutableMap<String, Int>
+            val sale = saleCount as MutableMap<String, Int>
+            for ((key, value) in quotasMap) {
+                val produceQuota = ProduceQuota(
+                    produceName = key,
+                    produceGoalAmount = value,
+                    saleAmount = sale.getOrDefault(key, 0)
+                )
+                produceList.add(produceQuota)
+            }
+            return Quota(
+                id = id,
+                produceQuotaList = produceList
+            )
 
-        return if (quotaIndex >= 0) {
-            quotasList[quotaIndex] = quotasList[quotaIndex].copy(produceQuotaList = produce)
-            null
         } else {
-            quotasList.add(Quota(
-                id = currentId,
-                produceQuotaList = produce,
-            ))
-            val tempId = currentId
-            currentId = (currentId.toInt() + 1).toString()
-            tempId
+            return null
         }
     }
+    fun addQuota(market: MarketModel.Market, produce: List<QuotasRepository.ProduceQuota>) {
+        val produceQuota = mutableMapOf<String, Int>()
+
+        for (prod in produce) {
+            produceQuota[prod.produceName] = prod.produceGoalAmount
+        }
+
+        val data = hashMapOf(
+            "produce" to produceQuota,
+            "sale" to SALES,
+        )
+        db.collection("quotas").document(market.id)
+            .set(data)
```

Comment:
You actually don't need to reset the document, you can just update the produce map. You can change all this code to just the following:

```
db.collection("quotas").document(market.id)
    .update("produce", produce.associate(
        it.produceName to it.produceGoalAmount
    ))
```

Reply:
Make sure this still works but it should be good. And then you can just add the onSucceessListner and stuff on top of it,

Reply:
Yup tested it it works :)